### PR TITLE
Support passing in a function to determine service name for mysql

### PR DESCRIPTION
**Pull request by jhatch**
### What does this PR do?
Similar to the existing behavior of the `OracleDB` package and others - this PR adds support for passing in a function to dynamically set `service` for the `mysql` and `mysql2` packages.

### Motivation
We have a service that queries multiple databases owned by other services (has to do with outbox event propagation). Let's call this service "something-api". Currently the spans encompassing queries from "something-api" to ANY database is "something-api-mysql". To ease debugging, we'd like to have a unique name per DB-instance we're querying. We could achieve this on our end if service was definable via a function we pass in. 

### Plugin Checklist

- [x] Unit tests.
- [x] TypeScript [definitions][1].
- [x] TypeScript [tests][2].
- [x] API [documentation][3].
- [x] CircleCI [jobs/workflows][4].
- [x] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/dd-trace-js/blob/master/docs/API.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

File: packages/datadog-plugin-mysql/src/index.js [hunk 1]
```diff
@@ -55,4 +56,14 @@ class MySQLPlugin extends Plugin {
   }
 }
 
+function getServiceName (tracer, config) {
+  if (typeof config.service === 'function') {
+    return config.service()
```
Comment (rochdev) [inline packages/datadog-plugin-mysql/src/index.js:61]: Should something be passed to this function? Otherwise, how can it decide what servie name to use?

Comment (jhatch) [reply packages/datadog-plugin-mysql/src/index.js]: Good call. In our case we are using async hooks to know the context we're in. But it makes more sense to simply pass the relevant config into the service function...updated. 👍

File: packages/datadog-plugin-mysql/src/index.js [hunk 2]
```diff
@@ -55,4 +56,14 @@ class MySQLPlugin extends Plugin {
   }
 }
 
+function getServiceName (tracer, config, dbConfig) {
+  if (typeof config.service === 'function') {
+    return config.service(dbConfig)
```
Comment (nampas) [inline packages/datadog-plugin-mysql/src/index.js:61]: Is the [documentation](https://datadoghq.dev/dd-trace-js/interfaces/plugins.mysql.html) generated from this repo? If so, would be good to update it to reflect that `service` can be a function.

Comment (rochdev) [reply packages/datadog-plugin-mysql/src/index.js:61]: Yes, as long as the API docs have been updated (which is the case for this PR) then the next time we release the documentation will be updated.

Comment (rochdev) [inline packages/datadog-plugin-mysql/src/index.js:61]: There should be a test for this.

**Review by rochdev — CHANGES_REQUESTED** [review]

File: packages/datadog-plugin-mysql/src/index.js (hunk 2 above)
Comment (jhatch) [reply packages/datadog-plugin-mysql/src/index.js:61]: Added tests to confirm this is called with correct parameters for both `mysql` and `mysql2`

**Review by rochdev — APPROVED** [review]

**Pull request merged**